assertSnapshot fails to match carriage return

**Describe the bug**
If you pass the CR character `\r` to `assertSnapshot`, it saves the snapshot fine but then immediately fails to match the snapshot.

I ran into this by trying to snapshot the output of `stringify(...)` from `https://deno.land/std@0.212.0/csv/mod.ts` which apparently output CRLFs (`\r\n`).

**Steps to Reproduce**


1. Create a file (`crlf.test.ts`) with the contents
    ```ts
    import { assertSnapshot } from 'https://deno.land/std@0.212.0/testing/snapshot.ts';
    
    Deno.test('CR failure', async (t) => {
      await assertSnapshot(t, '\r');
    });
   ```
2. Run `deno test --allow-read --allow-write cr.test.ts -- --update` which returns
    ```
    running 1 test from ./cr.test.ts
    CR failure ... ok (2ms)
    ------- output -------
    
    > 1 snapshot updated.
    
    ok | 1 passed | 0 failed (4ms)
    ```
    and saves `__snapshots__/cr.test.ts.snap` with 
    ```ts
    export const snapshot = {};
    
    snapshot[`CR failure 1`] = `
    "
    "
    `;
    ```
3. Run `deno test --allow-read cr.test.ts` which returns:
    ```
    running 1 test from ./cr.test.ts
    CR failure ... FAILED (6ms)
    
     ERRORS 
    
    CR failure => ./cr.test.ts:3:6
    error: AssertionError: Snapshot does not match:
    
    
        [Diff] Actual / Expected
    
    
    -   "
    +   "
        "
    
        throw new AssertionError(
              ^
        at assertSnapshot (https://deno.land/std@0.212.0/testing/snapshot.ts:591:11)
        at async file:///home/scotty/code/Toptal/FoodNoms/macrosai-supabase/cr.test.ts:4:3
    
     FAILURES 
    
    CR failure => ./cr.test.ts:3:6
    
    FAILED | 0 passed | 1 failed (8ms)
    
    error: Test failed
    ```

**Expected behavior**

The test should succeed after the snapshot has been updated.

**Environment**

<!-- please complete the following information -->

- OS: Ubuntu 22.04.3 LTS
- deno version: 1.39.1
- std version: 0.212.0

--- a/testing/snapshot.ts
+++ b/testing/snapshot.ts
@@ -222,7 +222,7 @@ export function serialize(actual: unknown): string {
     strAbbreviateSize: Infinity,
     breakLength: Infinity,
     escapeSequences: false,
-  }).replace(/\r/g, "\\r");
+  }).replaceAll("\r", "\\r");
 }
 
 /**

--- a/testing/snapshot.ts
+++ b/testing/snapshot.ts
@@ -222,7 +222,7 @@ export function serialize(actual: unknown): string {
     strAbbreviateSize: Infinity,
     breakLength: Infinity,
     escapeSequences: false,
-  });
+  }).replace(/\r/g, "\\r");
 }
 
 /**

--- a/testing/snapshot_test.ts
+++ b/testing/snapshot_test.ts
@@ -873,3 +873,7 @@ Deno.test(
     assert(!output.success, "The test should fail");
   }),
 );
+
+Deno.test("assertSnapshot() - should work with the string with '\\r' character", async (t) => {
+  await assertSnapshot(t, "Hello\r\nWorld!\r\n");
+});

